File: Programmers/Python/Lv.2/짝지어 제거하기.py
def solution(s):
    answer = -1
    stack = []
    s_list = list(s)
    

    for char in s_list :
        
        if len(stack) == 0 :
            stack.append(char)
        
        else :
            if stack[-1] == char :
                stack.pop()
            else :
                stack.append(char)
    
    if len(stack) == 0 :
        answer = 1
    else :
        answer = 0
    return answer

File: Programmers/Javascript/Lv.2/짝지어 제거하기.js
function solution(s) {
    const stack = [];

    for (let i = 0; i < s.length; i++) {
        if (i === 0) stack.push(s[i]);
        else {
            if (stack[stack.length - 1] === s[i]) stack.pop();
            else stack.push(s[i]);
        }
    }
    return !stack.length ? 1 : 0;
}
